src/extension/extension.ts

```diff
@@ -163,6 +163,12 @@ export function activate(context: vscode.ExtensionContext) {
 							placeHolder: 'Select a color',
 						}) as PetColor;
 						break;
+					case PetType.crab:
+						choices = [PetColor.red];
+						petColor = await vscode.window.showQuickPick(choices, {
```

Comment:
As the color can only be red, don't show a quick pick, just assign the color variable (see the duck and snake cases above)

Reply:
Good call, fixed!